calico/common.py

```diff
@@ -497,7 +547,21 @@ def validate_rules(profile_id, rules):
                 if tag is None:
                     continue
                 if not VALID_ID_RE.match(tag):
-                    issues.append("Invalid %s '%r'." % (tag_type, tag))
+                    issues.append("Invalid %s %r." % (tag_type, tag))
+
+            # For selectors, we replace the value with the parsed selector.
+            # This avoids having to re-parse it later and it ensures that
+            # equivalent selectors compare equal.
+            for sel_type in ('src_selector', 'dst_selector'):
+                sel_str = rule.get(sel_type)
+                if sel_str is None:
+                    continue
```

Comment:
No, we get None here if `sel_type` is not present, which is fine.  We strip out the Nones up above